zio-http/src/main/scala/zio/http/Response.scala

```diff
@@ -125,8 +121,9 @@ object Response {
     headers: Headers = Headers.empty,
     body: Body = Body.empty,
     httpError: Option[HttpError] = None,
+    encodedResponse: Option[Response.EncodedResponse] = None,
   ): Response =
-    Response(status, headers, body, Attribute.empty, httpError)
+    new Response(status, headers, body, Attribute.empty, httpError, new AtomicReference(None)) {}
```

Comment:
This allocates three 64 bit pointers as well as the atomic reference for each `Response`, when it is really only necessary for a cached response.

Reply:
`AtomicRef` removed, `Response` is back to being fairly pure now.